File: android/src/main/kotlin/com/michaellazebny/jyphoon/jc/utils/JCCallUtils.kt
package com.michaellazebny.jyphoon.jc.utils

import com.juphoon.cloud.*
import com.michaellazebny.jyphoon.jc.CallType
import com.michaellazebny.jyphoon.jc.jcWrapper.JCManager

object JCCallUtils {
    val activeCall: JCCallItem?
        get() {
            for (item in JCManager.getInstance().call.callItems) {
                if (item.active) {
                    return item
                }
            }
            return null
        }
    val otherParticipant: JCMediaChannelParticipant?
        get() {
            val jcManager = JCManager.getInstance()
            val selfParticipant = jcManager.mediaChannel?.selfParticipant
            var otherParticipant: JCMediaChannelParticipant? = null
            val participants = jcManager.mediaChannel.participants
            participants.forEach {
                if (it != selfParticipant) {
                    otherParticipant = it
                }
            }
            return otherParticipant
        }
    fun getCallType(): CallType? {
        return if (JCManager.getInstance().call.callItems.isNotEmpty()) {
            CallType.ONETOONE
        } else if (JCManager.getInstance().mediaChannel.selfParticipant != null) {
            CallType.GROUP
        } else {
            null
        }
    }
}

File: ios/Classes/utils/JCCallUtils.swift
import JCSDKOC

class JCCallUtils {

    static var otherParticipant: JCMediaChannelParticipant? {
        let jc = JCRoom.shared
        guard let selfParticipant = jc.mediaChannel.selfParticipant else { return nil }
        var otherParticipant: JCMediaChannelParticipant? = nil
        let participants = jc.mediaChannel.participants
        for participant in participants where participant as? JCMediaChannelParticipant != selfParticipant {
            otherParticipant = participant as? JCMediaChannelParticipant
            break
        }
        return otherParticipant
    }
    
    static func getCallType() -> CallType? {
        let jc = JCRoom.shared
        if !(jc.call.callItems?.isEmpty ?? true) {
            return .oneToOne
        } else if jc.mediaChannel.selfParticipant != nil {
            return .group
        } else {
            return nil
        }
    }
}
